protocols/kad/src/query.rs

```diff
@@ -127,6 +127,24 @@ impl QueryPool {
         id
     }
 
+    /// Adds a query to the pool that iterates towards the closest peers to the target.
+    /// With `num_results` dynamic specified.
+    pub(crate) fn add_iter_closest_num_results<T, I>(
```

Comment:
Can be consolidated with `add_iter_closest` by adding a `Option<NonZeroUsize>` parameter. This shouldn't be a problem since the function is only used inside of this crate.